packages/protocol/contracts/dapis/aggregators/Aggregator.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.2;
+
+/// @title The abstract contract that generalizes aggregation methods
+/// @notice Aggregation method implementations should inherit this contract
+abstract contract Aggregator {
+    /// @notice Called internally by the dAPI contract to aggregate an array of
+    /// values
+    /// @dev It is not guaranteed that `values` will not be changed after this
+    /// method runs
+    /// @param values Values to be aggregated
+    function aggregateInplace(int256[] memory values)
+        internal
+        view
```

Comment:
Had to switch from `pure` to `view` because `MeanMedianHybrid.sol` reads `toleranceInPercentages` from storage. A lot of the potential aggregator contracts such as weighted ones will also do that so this is probably fine.

Reply:
What if you set the parameter to a constant?

Reply:
I think tolerance should be modifiable (by the owner)